lib/spack/llnl/util/tty/log.py

```diff
@@ -13,12 +13,18 @@
 import select
 import sys
 import traceback
+import signal
 from contextlib import contextmanager
 from six import string_types
 from six import StringIO
 
 import llnl.util.tty as tty
 
+try:
+    import termios
+except:
+    termios = None
```

Comment:
This is messy.  Just move it to where it is used.  It is used in exactly one place, and I don't know why this needs to be here.  This *also* adds to the number of imports we need to do when we import this package.

Reply:
It's used in 3 places.

Reply:
ok actually, fair enough -- I might like this one.  Want to keep this one the way it is?  It does make the code clearer across `__enter__` and `__exit__`